### PR TITLE
Fix inexact in plasma flow types

### PR DESCRIPTION
**Pull request by maniator**
Updated version of eslint to support the new flow `...` inexact definition

Also updated the export flow-typed definitions to include the defs for all static variables and methos defined on all classes exported by plasma.

ESLint was updated to understand the new `...` syntax (together with fixing all the new lint errors)

### CODE REVIEW

File: src/components/TextInput/TextInput.jsx [hunk 1]
```diff
@@ -4,7 +4,7 @@ import cx from 'classnames';
 
 import { getDataAttrs } from '../../dataUtils';
 import type {
-  ChangeEventHanlders,
+  ChangeEventHandlers,
```
Comment (kangax) [inline src/components/TextInput/TextInput.jsx]: lol

File: flow/plasma_vx.x.x.js [hunk 2]
```diff
@@ -222,24 +220,26 @@ declare module '@wework-dev/plasma' {
...
-  declare type HandleEvent = {|
-    nativeEvent: { target: { value: any } },
-    target: { parentElement: { classList: { add: string => mixed, remove: string => mixed } } },
-  |};
+  declare type HandleEvent =
+    | {|
+        nativeEvent: { target: { value: any } },
+        target: { parentElement: { classList: { add: string => mixed, remove: string => mixed } } },
+      |}
+    | string;
```
Comment (kangax) [inline flow/plasma_vx.x.x.js]: is this correct? when would HandleEvent be a string?

File: src/components/NumberInput/NumberInput.jsx [hunk 3]
```diff
@@ -20,19 +20,19 @@ type HandleEvent = {|
       value: Object,
     },
   },
-|};
+|} | string;
```
Comment (kobim) [inline src/components/NumberInput/NumberInput.jsx]: I'm with @kangax here too.. we should understand where this is being used with a string.

File: src/types.js [hunk 4]
```diff
@@ -1,7 +1,5 @@
 // @flow
-export type Data = {
-  [key: string]: string,
-};
+export type Data = { [key: string]: string, ... };
```
Comment (kobim) [inline src/types.js]: I think this should be an exact type

File: src/types.js [hunk 5]
```diff
@@ -67,4 +65,4 @@ export type FocusEventHandlers<T: EventTarget = EventTarget> = {|
   onBlur?: SyntheticFocusEventHandler<T>,
 |};
 
-export type StyleAttributes = { [string]: string | number };
+export type StyleAttributes = { [string]: string | number, ... };
```
Comment (kobim) [inline src/types.js]: maybe `{ [key: string]: string | number | { ... } }`? (as `StyleAttributes` will always have string keys.)

File: flow/plasma_vx.x.x.js (hunk 2 above)
Comment (maniator) [reply flow/plasma_vx.x.x.js]: We are passing it a string in one place in plasma it seems

I can look more tomorrow when I am by my computer

File: src/types.js (hunk 4 above)
Comment (kobim) [reply src/types.js]: ping ^^

**Comment by kangax** [comment]
@kobim we need to tell Flow about React components' default props as it influences the way Flow sees props internally (used in a component) vs. externally (when component is used somewhere else) — https://flow.org/en/docs/react/components/#toc-using-default-props

Basically, default prop in Flow is marked as a non-optional property and it knows to not require it when the component is used somewhere else AND also not error out when you do smth like `this.props.propThatHasDefaultValue()` inside the component.

Hope that makes sense.

**Pull request merged**